Sources/Document/CRDT/RHTPQMap.swift

```diff
@@ -0,0 +1,219 @@
+        guard let node = self.nodeMapByCreatedAt[createdAt] else {
+            let log = "can't find the given node: \(createdAt)"
+            Logger.fatal(log)
+            throw YorkieError.unexpected(message: log)
+        }
+
+        return node.rhtKey
+    }
+
+    /**
+     * `delete` physically deletes child element.
+     */
+    func delete(value: CRDTElement) throws {
+        guard let node = self.nodeMapByCreatedAt[value.getCreatedAt()] else {
+            let log = "can't find the given node: \(value.getCreatedAt())"
+            Logger.fatal(log)
+            throw YorkieError.unexpected(message: log)
+        }
+
+        guard let queue = self.elementQueueMapByKey[node.rhtKey] else {
+            let log = "can't find the given node: \(node.rhtKey)"
+            Logger.fatal(log)
+            throw YorkieError.unexpected(message: log)
+        }
+
+        let heapNode = HeapNode(key: node.rhtValue.getCreatedAt(), value: node)
+        queue.delete(heapNode)
+        self.nodeMapByCreatedAt[node.rhtValue.getCreatedAt()] = nil
+    }
+
+    /**
+     * `remove` removes the Element of the given key and removed time.
+     */
+    func remove(key: String, executedAt: TimeTicket) throws -> CRDTElement {
+        guard let heap = self.elementQueueMapByKey[key],
+              let node = heap.peek()
+        else {
+            let log = "can't find the given node: \(key)"
+            Logger.fatal(log)
+            throw YorkieError.unexpected(message: log)
+        }
+
+        node.value.remove(removedAt: executedAt)
+        return node.value.rhtValue
+    }
+
+    /**
+     * `has` returns whether the element exists of the given key or not.
+     */
+    func has(key: String) -> Bool {
+        guard let heap = self.elementQueueMapByKey[key],
+              let node = heap.peek()
+        else {
+            return false
+        }
+
+        return node.value.isRemoved() == false
+    }
+
+    /**
+     * `get` returns the value of the given key.
+     */
+    func get(key: String) throws -> CRDTElement {
+        guard let heap = elementQueueMapByKey[key], let node = heap.peek() else {
+            let log = "can't find the given node: \(key)"
+            Logger.fatal(log)
+            throw YorkieError.unexpected(message: log)
+        }
+
+        return node.value.rhtValue
+    }
+}
+
+extension RHTPQMap: Sequence {
+    typealias Element = RHTPQMapNode
+
+    func makeIterator() -> RHTPQMapIterator {
+        return RHTPQMapIterator(self.elementQueueMapByKey)
+    }
+}
+
+class RHTPQMapIterator: IteratorProtocol {
+    private var target: [Heap<TimeTicket, RHTPQMapNode>]
+    private var currentNodes: [RHTPQMapNode] = []
+
+    init(_ target: [String: Heap<TimeTicket, RHTPQMapNode>]) {
+        self.target = Array(target.values)
+    }
+
+    func next() -> RHTPQMapNode? {
+        while true {
+            guard self.currentNodes.isEmpty else {
+                break
```

Comment:
```suggestion
                return nil
```

Reply:
It should return the result of `self.currentNodes.removeFirst()`.

Reply:
I misread guard as if. Sorry.